emcc.py

```diff
@@ -1861,6 +1861,8 @@ def default_setting(name, new_default):
       exit_with_error('USE_PTHREADS=2 is no longer supported')
     if settings.ALLOW_MEMORY_GROWTH:
       diagnostics.warning('pthreads-mem-growth', 'USE_PTHREADS + ALLOW_MEMORY_GROWTH may run non-wasm code slowly, see https://github.com/WebAssembly/design/issues/1271')
+    if settings.BUILD_AS_WORKER:
+      diagnostics.warning('pthreads-build-as-worker', 'USE_PTHREADS + BUILD_AS_WORKER require separate modes that don\'t work together, see https://github.com/emscripten-core/emscripten/issues/8854')
```

Comment:
How about we just `exit_with_error` here since they are know to be incompatible.

Reply:
I'm not immediately opposed to that.

My reason for not is that it takes situations that previously gave output that mostly worked (so long as they don't actually use threads despite enabling them) and intentionally tanks them. No matter how well intentioned the cause, that's not necessarily a fun experience.

I don't know what messy build systems people are dealing with in the wild, but I would rather let there be a warning that lets people know there is a problem, but proceeds anyway rather than force them to wrangle with the build system any further than is absolutely necessary.

Reply:
In any case, I've made the suggested changes